pallets/settlement/src/lib.rs

```diff
@@ -842,55 +716,58 @@ decl_module! {
+        /// * `offchain_transfers` - The number of offchain assets in the instruction.
         ///
         /// # Permissions
         /// * Portfolio
-        #[weight = <T as Config>::WeightInfo::reject_instruction_v2(*fungible_transfers, *nfts_transfers)]
-        pub fn reject_instruction_v2(origin, id: InstructionId, portfolio: PortfolioId, fungible_transfers: u32, nfts_transfers: u32) -> DispatchResult {
-            Self::base_reject_instruction(origin, id, portfolio, fungible_transfers, Some(nfts_transfers))
+        #[weight = <T as Config>::WeightInfo::reject_instruction(*fungible_transfers, *nfts_transfers, *offchain_transfers)]
+        pub fn reject_instruction(
+            origin,
+            id: InstructionId,
+            portfolio: PortfolioId,
+            fungible_transfers: u32,
+            nfts_transfers: u32,
+            offchain_transfers: u32
+        ) -> DispatchResult {
+            let input_cost = TransferData::new(fungible_transfers, nfts_transfers, offchain_transfers);
+            Self::base_reject_instruction(origin, id, portfolio, &input_cost)
         }
 
         /// Root callable extrinsic, used as an internal call to execute a scheduled settlement instruction.
-        #[weight = <T as Config>::WeightInfo::execute_scheduled_instruction(*fungible_transfers, *nfts_transfers)]
+        #[weight = <T as Config>::WeightInfo::execute_scheduled_instruction(*fungible_transfers, *nfts_transfers, 0)]
         fn execute_scheduled_instruction_v2(
             origin,
             id: InstructionId,
             fungible_transfers: u32,
             nfts_transfers: u32
-        ) -> DispatchResultWithPostInfo {
+        ) -> DispatchResultWithPostInfo{
             Self::ensure_root_origin(origin)?;
             let mut weight_meter = Self::ensure_valid_weight_meter(
                 Self::execute_scheduled_instruction_minimum_weight(),
-                Self::execute_scheduled_instruction_weight_limit(fungible_transfers, nfts_transfers),
+                Self::execute_scheduled_instruction_weight_limit(fungible_transfers, nfts_transfers, 0),
             )?;
-            Self::base_execute_scheduled_instruction(id, &mut weight_meter).map_err(|e| {
-                DispatchErrorWithPostInfo {
-                    post_info: Some(weight_meter.consumed()).into(),
-                    error: e.error,
-                }
-            })
+            Ok(Self::base_execute_scheduled_instruction(id, &mut weight_meter))
         }
 
         /// Root callable extrinsic, used as an internal call to execute a scheduled settlement instruction.
-        #[weight = (*weight_limit).max(<T as Config>::WeightInfo::execute_scheduled_instruction(0, 0))]
-        fn execute_scheduled_instruction_v3(origin, id: InstructionId, weight_limit: Weight) -> DispatchResultWithPostInfo {
+        #[weight = (*weight_limit).max(<T as Config>::WeightInfo::execute_scheduled_instruction(0, 0, 0))]
+        fn execute_scheduled_instruction_v3(
```

Comment:
We can change the names of these `execute_schedule_instruction*` extrinsics, just not change the order or parameters of the old ones.

But we can do that cleanup in a different PR.  I created MESH-2047 for adding a migration to remove the old schedule extrinsics.